dts/arm/nordic/nrf51822.dtsi

```diff
@@ -5,6 +5,15 @@
 #include <dt-bindings/gpio/gpio.h>
 #include "nrf5_common.dtsi"
 
+#include <nordic/override.h>
+/*
+ * This value can be overridden at the board level or in an application specific
+ * dts/nordic/override.h file.
+ */
+#ifndef DT_DEFAULT_IRQ_PRIORITY
+#define DT_DEFAULT_IRQ_PRIORITY 1
```

Comment:
Use the lowest priority as default, which is 3 IIRC for nRF51

Reply:
I am not sure we should do this globally, as this can potentially affect existing applications. Currently this PR only makes changing of default priority easier, the existing values are not modified, except for GPIOTE, which for unknown reason had priority 5 set on nRF52, and that got propagated on nRF91 and nRF53 (on nRF51 it was 1, not 3).
Perhaps the priorities should be lowered only where it is actually needed, e.g. in BLE-related samples, as suggested by @ioannisg in https://github.com/zephyrproject-rtos/zephyr/pull/19703#issuecomment-542133475.

What are the others' opinions regarding this?

Reply:
> potentially affect existing applications

This is not an acceptable reasoning for me to select a default priority that is not highest or lowest in an SoC.

- Defaults can be to have highest priority and the use of Bluetooth controller would then change the defaults of all IRQs except the ones used by Bluetooth to a lower value. This is even more confusing to user of other peripherals.
- can be to have lowest priority and only the Bluetooth controller used IRQs are set to the required priority